tools/code_analyzer/op_dependency.cpp

```diff
@@ -0,0 +1,743 @@
+//  2) one possible call path to at::empty() function;
+//  3) the called operator name "aten::empty":
+//
+// - quantized::add
+// - c10::detail::wrap_kernel_functor_unboxed_<at::native::(anonymous
+//   namespace)::QAdd<false>, at::Tensor (at::Tensor, at::Tensor, double,
+//   long)>::call(c10::OperatorKernel*, at::Tensor, at::Tensor, double, long)
+// - at::native::(anonymous namespace)::QAdd<false>::operator()(at::Tensor,
+//   at::Tensor, double, long)
+// - void at::native::DispatchStub<void (*)(at::Tensor&, at::Tensor const&,
+//   at::Tensor const&), at::native::qadd_stub>::operator()<at::Tensor&,
+//   at::Tensor const&, at::Tensor const&>(c10::DeviceType, at::Tensor&,
+//   at::Tensor const&, at::Tensor const&)
+// - at::native::DispatchStub<void (*)(at::Tensor&, at::Tensor const&,
+//   at::Tensor const&), at::native::qadd_stub>::choose_cpu_impl()
+// - void at::native::(anonymous namespace)::qadd_kernel<false>(at::Tensor&,
+//   at::Tensor const&, at::Tensor const&)
+// - at::TensorIterator::binary_op(at::Tensor&, at::Tensor const&, at::Tensor
+//   const&, bool)
+// - at::TensorIterator::build()
+// - at::TensorIterator::fast_set_up()
+// - at::empty(c10::ArrayRef<long>, c10::TensorOptions const&,
+//   c10::optional<c10::MemoryFormat>)
+// - aten::empty
+
+#include <deque>
+#include <iostream>
+#include <set>
+#include <unordered_map>
+#include <unordered_set>
+
+#include "llvm/Demangle/Demangle.h"
+#include "llvm/Analysis/LazyCallGraph.h"
+#include "llvm/IR/CallSite.h"
+#include "llvm/IR/Constant.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/Instruction.h"
+#include "llvm/IR/Intrinsics.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Pass.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/Regex.h"
+#include "llvm/Support/raw_ostream.h"
+
+using namespace llvm;
+
+namespace {
+
+struct RegexOpt {
+  std::shared_ptr<Regex> pattern;
+
+  void operator=(const std::string& val) {
+    if (val.empty()) {
+      return;
+    }
+    pattern = std::make_shared<Regex>(val);
+    std::string regexError;
+    if (!pattern->isValid(regexError)) {
+      report_fatal_error(
+          "Invalid regular expression param: '" + val + "' err: " + regexError,
+          false);
+    }
+  };
+};
+
+static RegexOpt FunctionSchemaPatternLoc;
+static cl::opt<RegexOpt, true, cl::parser<std::string>> FunctionSchemaPattern(
+    "op_schema_pattern",
+    cl::desc("Regular expression used to identify aten op schema strings. "
+             "Example: -op_schema_pattern '^(aten|quantized)::[^ ]+'"),
+    cl::location(FunctionSchemaPatternLoc),
+    cl::Required,
+    cl::ValueRequired);
+
+static RegexOpt OpRegistrationPatternLoc;
+static cl::opt<RegexOpt, true, cl::parser<std::string>> OpRegistrationPattern(
+    "op_register_pattern",
+    cl::desc("Regular expression used to identify c10 op registration API. "
+             "Example: -op_register_pattern 'c10::RegisterOperators::op'"),
```

Comment:
This can also be `torch::RegisterOperators`

Reply:
Are they type alias? I did a search on the llvm assembly - seems all of them were c10::RegisterOperators. I could add this as well to be safe.